### PR TITLE
[PT FE] Support aten::atan2 for pytorch models

### PR DESCRIPTION
**Pull request by tiebreaker4869**
### Details:
- add atan2 operator and unit tests

### Tickets:
 - [[Good First Issue]: Support aten::atan2 for pytorch models](https://github.com/openvinotoolkit/openvino/issues/20575)

### CODE REVIEW

**Comment by rkazants** [comment]
build_jenkins

**Comment by mlukasze** [comment]
clang format check fails, fix it, please

File: src/frontends/pytorch/src/op/atan2.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    // do adjustment
+    auto atan_plus_pi = context.mark_node(std::make_shared<v1::Add>(atan, pi));
+    auto atan_minus_pi = context.mark_node(std::make_shared<v1::Subtract>(atan, pi));
+
+    // select result
+    auto adjusted_atan = context.mark_node(std::make_shared<v1::Select>(
+        x_greater_than_zero,
+        atan,
+        context.mark_node(std::make_shared<v1::Select>(add_pi_condition, atan_plus_pi, atan_minus_pi))));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/atan2.cpp:68]: Please create this operation on a separate line.

File: src/frontends/pytorch/src/op/atan2.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    std::tie(lhs, rhs) = get_inputs_with_promoted_types(context, 0, 1);
+
+    auto div = context.mark_node(std::make_shared<v1::Divide>(lhs, rhs));
+
+    auto atan = context.mark_node(std::make_shared<v0::Atan>(div));
+
+    // create some constants to adjust result according to quadrant.
+    auto zero = context.mark_node(v0::Constant::create(rhs.get_element_type(), Shape{}, {0}));
+    auto pi = context.mark_node(v0::Constant::create(rhs.get_element_type(), Shape{}, {M_PI}));
+    auto neg_pi = context.mark_node(v0::Constant::create(rhs.get_element_type(), Shape{}, {-M_PI}));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/atan2.cpp:46]: `rhs` or `lhs` type can be `dynamic` in that case creation of constant will fail. Please create constant with `i32` or `f64` type and use `ConvertLike` to make it of right dtype.

**Review by mvafin — CHANGES_REQUESTED** [review]

File: src/frontends/pytorch/src/op/atan2.cpp (hunk 2 above)
Comment (tiebreaker4869) [reply src/frontends/pytorch/src/op/atan2.cpp]: i have made the change. please take a look when you have a moment :)

File: src/frontends/pytorch/src/op/atan2.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    // create some constants to adjust result according to quadrant.
+    auto zero = context.mark_node(v0::Constant::create(ov::element::i32, Shape{}, {0}));
+    auto pi = context.mark_node(v0::Constant::create(ov::element::f64, Shape{}, {M_PI}));
+    auto neg_pi = context.mark_node(v0::Constant::create(ov::element::f64, Shape{}, {-M_PI}));
+    auto half_pi = context.mark_node(v0::Constant::create(ov::element::f64, Shape{}, {M_PI_2}));
+    auto neg_half_pi = context.mark_node(v0::Constant::create(ov::element::f64, Shape{}, {-M_PI_2}));
+
+    zero = context.mark_node(std::make_shared<v1::ConvertLike>(zero, rhs));
+    pi = context.mark_node(std::make_shared<v1::ConvertLike>(pi, rhs));
+    neg_pi = context.mark_node(std::make_shared<v1::ConvertLike>(neg_pi, rhs));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/atan2.cpp:57]:
```suggestion
    auto half_pi = context.mark_node(v0::Constant::create(ov::element::f64, Shape{}, {M_PI_2}));
    auto neg_half_pi = context.mark_node(v0::Constant::create(ov::element::f64, Shape{}, {-M_PI_2}));

    zero = context.mark_node(std::make_shared<v1::ConvertLike>(zero, rhs));
    pi = context.mark_node(std::make_shared<v1::ConvertLike>(pi, rhs));
```
You are not using `neg_pi` in your code.

**Review by mvafin — CHANGES_REQUESTED** [review]
In general looks good

**Comment by mvafin** [comment]
Tests for torch.export are failing. Please add `aten.atan2.default` to fx ops table

**Comment by tiebreaker4869** [comment]
hi @mvafin are there any further changes i need to make? :)

**Comment by mvafin** [comment]
build_jenkins

**Review by mvafin — APPROVED** [review]
@tiebreaker4869 Thank you for contribution

**Pull request merged**